Implement groupByNodes() function

ctrochalakis opened the pull request:
This is essentially a more powerful version of `groupByNode()` that allows grouping by multiple nodes.

We are evaluating graphite (through grafana) by producing various graphs based on our metrics and there some cases where we need to group a series list by multiple nodes. The tests should be clear on when that might be useful.

What concerns me is that we introduce a new version of an already present function, I am not sure if we can extend the functionally of `groupByNode()` to support multiple nodes. That could get tricky because of the parameter order and that might be difficult to express to other libraries that try to autocomplete graphite functions (like grafana).

The first commit adds tests for `groupByNode()` and can be merged as-is to master if you find it ok, I can also create a separate PR for it.

bmhatfield commented:
@ctrochalakis thanks for this PR!

You might not be surprised to learn that there's been a few PRs around this.

In previous discussions, I think we'd agreed https://github.com/graphite-project/graphite-web/pull/1171 was going to be the merged version, but our fearless PR author has not returned to make it mergeable. I personally like the name "groupByNodes" better than the longer "multi" versions.

ctrochalakis commented:
@bmhatfield I started implementing it and never checked for other PRs, sorry for that :)

ctrochalakis commented:
@bmhatfield I hijacked #1171, rebased to master and changed the documentation a bit (the `as defined by multiple nodes` part). I have kept the original author's name.

I pushed it here since I couldn't update #1171.

obfuscurity commented:
This PR now officially supercedes #1171. Adding @gboily's original description here for context.

```
Add a new function 'groupByNodes' that support multiple indexes like the function 'aliasByNode'.

This avoids to add a new graph target for each new metric/node created.

Ex:
&target=groupByNodes(ganglia.**server`*`**.`*`.cpu.**load`*`**,"sumSeries",1,4)

Will "generate" the following formula : 
- sumSeries(ganglia.server1.*.cpu.load5)
- sumSeries(ganglia.server1.*.cpu.load10)
- sumSeries(ganglia.server1.*.cpu.load15)
- sumSeries(ganglia.server2.*.cpu.load5)
- sumSeries(ganglia.server2.*.cpu.load10)
- sumSeries(ganglia.server2.*.cpu.load15)
- ...

Will produce the following legends : 
- server1.load5
- server1.load10
- server1.load15
- ...
- server2.load5
- server2.load10
- server2.load15
- ...
```

bmhatfield commented:
@ctrochalakis Thank you so much for doing this!!

From my perspective, this looks like a nice improvement and low risk, and something we've been waiting to merge for some time. @obfuscurity I am good to merge this unless there's any reservations?

obfuscurity commented:
:shipit:
